pkg/scheduler/framework/plugins/defaultpreemption/default_preemption_test.go

```diff
@@ -927,6 +927,40 @@ func TestPickOneNodeForPreemption(t *testing.T) {
 	}
 }
 
+func TestPodEligibleToPreemptOthers(t *testing.T) {
+	tests := []struct {
+		name          string
+		pod           *v1.Pod
+		pods          []*v1.Pod
+		nodes         []string
+		nodesStatuses framework.NodeToStatusMap
+		expected      bool
+	}{
+		{
+			name:  "Pod with nominated node",
+			pod:   &v1.Pod{ObjectMeta: metav1.ObjectMeta{Name: "machine1", UID: types.UID("machine1")}, Spec: v1.PodSpec{Containers: smallContainers}, Status: v1.PodStatus{NominatedNodeName: "machine1"}},
+			pods:  []*v1.Pod{},
+			nodes: []string{"machine1"},
+			nodesStatuses: framework.NodeToStatusMap{
+				"machine1": framework.NewStatus(framework.UnschedulableAndUnresolvable, tainttoleration.ErrReasonNotMatch),
+			},
+			expected: true,
+		},
+	}
```

Comment:
I'd like to expect another test to cover`expected = false`.

Reply:
Since there are missing unittests for podEligibleToPreemptOther method, I orginially plan to add more tests in another PR, but I think I can add them in this PR with separated commit. Let me know if it is good way your expected.

Reply:
SG.